src/net.cpp

```diff
@@ -2185,18 +2305,33 @@ uint64_t CNode::GetOutboundTargetBytesLeft()
     return (nMaxOutboundTotalBytesSentInCycle >= nMaxOutboundLimit) ? 0 : nMaxOutboundLimit - nMaxOutboundTotalBytesSentInCycle;
 }
 
-uint64_t CNode::GetTotalBytesRecv()
+uint64_t CConnman::GetTotalBytesRecv()
 {
     LOCK(cs_totalBytesRecv);
     return nTotalBytesRecv;
 }
 
-uint64_t CNode::GetTotalBytesSent()
+uint64_t CConnman::GetTotalBytesSent()
 {
     LOCK(cs_totalBytesSent);
     return nTotalBytesSent;
 }
 
+uint64_t CConnman::GetLocalServices() const
+{
+    return nLocalServices;
+}
+
+void CConnman::SetBestHeight(int height)
```

Comment:
Yea, I don't believe this deviates from the current behavior much, other than slightly obfuscating the exact current known height, since the value is grabbed at connection time now rather than during the handshake.

Though there's the advantage of using the accurate cached value based on a push event, rather than requiring a lock-and-pull as before. I don't think there's any need to try to replicate that old behavior here.